.github/workflows/scan-image-grype.yml

```
@@ -0,0 +1,190 @@
+            echo "Error: Invalid timeout-minutes value '${{ inputs.timeout-minutes }}'"
+            echo "Value must be a number between 1 and 360 minutes"
+            exit 1
+          fi
+      
+      - name: Validate retention-days
+        run: |
+          if [[ ! "${{ inputs.retention-days }}" =~ ^[0-9]+$ ]] || (( "${{ inputs.retention-days }}" < 1 )) || (( "${{ inputs.retention-days }}" > 90 )); then
+            echo "Error: Invalid retention-days value '${{ inputs.retention-days }}'"
+            echo "Value must be a number between 1 and 90 days"
+            exit 1
+          fi
+      
+      - name: Validate category-prefix
+        run: |
+          if [[ -z "${{ inputs.category-prefix }}" ]]; then
+            echo "Error: category-prefix cannot be empty"
+            exit 1
+          fi
+          if [[ "${{ inputs.category-prefix }}" =~ [^a-zA-Z0-9\-_] ]]; then
+            echo "Error: category-prefix can only contain alphanumeric characters, hyphens, and underscores"
+            exit 1
+          fi
+
+  scan:
+    name: Scan Image Grype SARIF
+    needs: validate
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+    concurrency:
+      group: ${{ inputs.image }}
+      cancel-in-progress: false
+    permissions:
+      security-events: write  # Needed to upload SARIF results
+      contents: read         # Needed to read workflow files
+    
+    steps:
+      - name: Extract image details
+        id: image_details
+        run: |
+          IMAGE_NAME=$(echo "${{ inputs.image }}" | cut -d':' -f1)
+          IMAGE_TAG=$(echo "${{ inputs.image }}" | cut -d':' -f2)
+          [[ "$IMAGE_TAG" == "$IMAGE_NAME" ]] && IMAGE_TAG="latest"
+          SAFE_NAME=$(echo "${IMAGE_NAME}-${IMAGE_TAG}" | sed 's/[\/:]/-/g')
+          {
+            echo "image_name=${IMAGE_NAME}"
+            echo "image_tag=${IMAGE_TAG}"
+            echo "safe_name=${SAFE_NAME}"
+          } >> "$GITHUB_OUTPUT"
+      
+      - name: Scan image with Grype
+        uses: anchore/scan-action@7c05671ae9be166aeb155bad2d7df9121823df32
+        id: scan
+        continue-on-error: true  # Allow workflow to continue even if scan fails
+        with:
+          image: "${{ inputs.image }}"
+          fail-build: "${{ inputs.fail-build }}"
+          severity-cutoff: "${{ inputs.severity-cutoff }}"
+          output-format: sarif
+          output-file: "${{ inputs.output-file }}"
+          by-cve: true
+      
+      - name: Check scan status
+        if: steps.scan.outcome == 'failure'
+        run: |
+          echo "::warning::Scan failed for image ${{ inputs.image }}"
+          echo "Please check the scan logs above for details"
+          if [ "${{ inputs.fail-build }}" = "true" ]; then
+            echo "::error::Build will fail due to scan failure and fail-build=true"
+            exit 1
+          fi
+      
+      - name: Enrich or generate SARIF
+        if: always()
```

```suggestion
        if: !cancelled()
```

This will allow the user to cancel the run, say if for some reason it is stuck